server/controllers/stock/reports/stock/lots_report.js

```diff
@@ -52,8 +55,48 @@ function stockLotsReport(req, res, next) {
   options.month_average_consumption = req.session.stock_settings.month_average_consumption;
   options.average_consumption_algo = req.session.stock_settings.average_consumption_algo;
 
+  const purgeKeys = ['NO_CONSUMPTION', 'S_MONTH', 'S_RISK', 'S_RISK_QUANTITY',
+    'at_risk_of_stock_out', 'cmms', 'color',
+    'default_purchase_interval', 'delay', 'depot_uuid',
+    'enterprisePurchaseInterval', 'exhausted', 'expired',
+    'inventory_uuid', 'lifetime_lot', 'min_delay',
+    'min_months_security_stock', 'mvt_quantity', 'near_expiration',
+    'purchase_interval', 'tag_name', 'tracking_consumption',
+    'tracking_expiration', 'wac'];
+
+  const dateKeys = ['min_stock_date', 'max_stock_date'];
+
+  // stock status label keys
+  // WARNING: Must match stockStatusLabelKeys in client StockService
+  // TODO: Move to shared constants file for client and server sides
+  const stockStatusLabelKeys = {
+    stock_out         : 'STOCK.STATUS.STOCK_OUT',
+    in_stock          : 'STOCK.STATUS.IN_STOCK',
+    security_reached  : 'STOCK.STATUS.SECURITY',
+    minimum_reached   : 'STOCK.STATUS.MINIMUM',
+    over_maximum      : 'STOCK.STATUS.OVER_MAX',
+    unused_stock      : 'STOCK.STATUS.UNUSED_STOCK',
+  };
+
   return Stock.getLotsDepot(null, options)
     .then((rows) => {
+      // Purge unneeded fields from the row
+      rows.forEach(row => {
```

Comment:
Just FYI - lodash has a utility for this:
```js
const rows = _.map(rows, row => _.omit(keys, row));
```

Reply:
Actually I don't think this will work.  _.omit() creates a copy of the object with the rows removed.   I think the invocation as shown above needs to do the changes in place.  And the _.unset() function takes only one path, so you would have to do a two-level mapping.  I think the forEach code I have above is cleaner and more efficient than copying lots (omit) or by doing two-level maps (unset).